server/service/apple_mdm.go

```diff
@@ -1102,6 +1104,117 @@ func (svc *Service) ListMDMAppleInstallers(ctx context.Context) ([]fleet.MDMAppl
 	return installers, nil
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Batch Replace MDM Apple Profiles
+////////////////////////////////////////////////////////////////////////////////
+
+type batchSetMDMAppleProfilesRequest struct {
+	TeamID   *uint    `json:"-" query:"team_id,optional"`
+	TeamName *string  `json:"-" query:"team_name,optional"`
+	DryRun   bool     `json:"-" query:"dry_run,optional"` // if true, apply validation but do not save changes
+	Profiles [][]byte `json:"profiles"`
+}
+
+type batchSetMDMAppleProfilesResponse struct {
+	Err error `json:"error,omitempty"`
+}
+
+func (r batchSetMDMAppleProfilesResponse) error() error { return r.Err }
+
+func batchSetMDMAppleProfilesEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*batchSetMDMAppleProfilesRequest)
+	if err := svc.BatchSetMDMAppleProfiles(ctx, req.TeamID, req.TeamName, req.Profiles, req.DryRun); err != nil {
+		return batchSetMDMAppleProfilesResponse{Err: err}, nil
+	}
+	return batchSetMDMAppleProfilesResponse{}, nil
+}
+
+func (svc *Service) BatchSetMDMAppleProfiles(ctx context.Context, tmID *uint, tmName *string, profiles [][]byte, dryRun bool) error {
+	if !svc.config.MDMApple.Enable {
```

Comment:
For now I required that this was set. As commented below, I suspect this will become just a check that APNs is configured when we remove the feature flag.